crates/bevy_animation/src/gltf_curves.rs

```diff
@@ -0,0 +1,420 @@
+                let interpolated =
+                    u.iter()
+                        .zip(v.iter())
+                        .map(move |(x, y)| if s >= 1.0 { y.clone() } else { x.clone() });
+                TwoIterators::Right(interpolated)
+            }
+        }
+    }
+}
+
+impl<T> WideSteppedKeyframeCurve<T> {
+    /// Create a new [`WideSteppedKeyframeCurve`]. An error will be returned if:
+    /// - `values` has length zero.
+    /// - `times` has less than `2` unique valid entries.
+    /// - The length of `values` is not divisible by that of `times` (once sorted, filtered,
+    ///   and deduplicated).
+    #[inline]
+    pub fn new(
+        times: impl IntoIterator<Item = f32>,
+        values: impl IntoIterator<Item = T>,
+    ) -> Result<Self, WideKeyframeCurveError> {
+        Ok(Self {
+            core: ChunkedUnevenCore::new_width_inferred(times, values)?,
+        })
+    }
+}
+
+/// A keyframe-defined curve that uses cubic interpolation over many samples at once, backed by a
+/// contiguous buffer.
+#[derive(Debug, Clone, Reflect)]
+pub struct WideCubicKeyframeCurve<T> {
+    core: ChunkedUnevenCore<T>,
+}
+
+impl<T> IterableCurve<T> for WideCubicKeyframeCurve<T>
+where
+    T: VectorSpace,
+{
+    #[inline]
+    fn domain(&self) -> Interval {
+        self.core.domain()
+    }
+
+    fn sample_iter_unchecked(&self, t: f32) -> impl Iterator<Item = T> {
+        match self.core.sample_interp_timed(t) {
+            InterpolationDatum::Exact((_, v))
+            | InterpolationDatum::LeftTail((_, v))
+            | InterpolationDatum::RightTail((_, v)) => {
+                // Pick out the part of this that actually represents the position (instead of tangents),
+                // which is the middle third.
+                let width = self.core.width();
+                TwoIterators::Left(v[width..(width * 2)].iter().copied())
+            }
+
+            InterpolationDatum::Between((t0, u), (t1, v), s) => TwoIterators::Right(
+                cubic_spline_interpolate_slices(self.core.width() / 3, u, v, s, t1 - t0),
+            ),
+        }
+    }
+}
+
+/// An error indicating that a multisampling keyframe curve could not be constructed.
+#[derive(Debug, Error)]
+#[error("Unable to construct a curve using this data")]
+pub enum WideKeyframeCurveError {
+    /// The number of given values was not divisible by a multiple of the number of keyframes.
+    #[error("The number of values ({values_given}) was expected to be divisible by {divisor}")]
+    LengthMismatch {
+        /// The number of values given.
+        values_given: usize,
+        /// The number that `values_given` was supposed to be divisible by.
+        divisor: usize,
+    },
+
+    /// An error was returned by the internal core constructor.
+    CoreError(#[from] ChunkedUnevenCoreError),
+}
+
+impl<T> WideCubicKeyframeCurve<T> {
+    /// Create a new [`WideCubicKeyframeCurve`]. An error will be returned if:
```

Comment:
```suggestion
    /// Create a new [`WideCubicKeyframeCurve`].
    ///
    /// An error will be returned if:
```
keeps the first line short and descriptive, so that when it appears as a listing in rustdoc, the "an error" part won't also be visible